test/LoggerTest.php

```diff
@@ -0,0 +1,215 @@
+
+        // create some messages and levels
+        $level1 = new LogLevel(Horde_Log::ALERT, 'Alert');
+        $message1 = 'this is a valid message for logmethodscheck';
+
+        // put all the methods fromt the LoggerInterface::class in an array
+        $loggerinterface_methods = get_class_methods(LoggerInterface::class);
+
+        // check for each method that the correct name, message and level are set.
+        // Because $mockhandler1 is loaded into $logger, $mockhandler1->check should containt what is logged through $logger->log($somelevel, $somemessage)
+        foreach ($loggerinterface_methods as $key => $method) {
+            if ($method == 'log') {
+                $logger->$method($level1, "logmethodscheck: for the log method");
+                $this->assertEquals($mockhandler1->check->message(), "logmethodscheck: for the log method");
+                $this->assertEquals($mockhandler1->check->level()->name(), $level1->name());
+            } else {
+                $logger->$method($message1);
+                $this->assertEquals($mockhandler1->check->message(), $message1);
+                $this->assertEquals($mockhandler1->check->level()->name(), $method);
+            }
+        }
+    }
+
+    /*
+    * Test if the log() function can pass a string instead of an instance of LogMessage or Stringable
+    * - Uses the default Logger (in SetUp()).
+    *
+    * Note:
+    * - default Logger uses a mockhandler1 and an additional filter with "emergency"
+    * - mockhandler1 will log a message if the $this->logging->log() method is called with "emergency" in it
+    * - mockhandler1->check will not hold the value "filtered out" because the filter is defined in the logger not in the handler
+    */
+    public function testStringAsMessageForLog()
+    {
+        $message = "emergency: this is a a message for an emergency!";
+        $this->logging->log($this->level1, $message);
+        $this->assertEquals($this->mockhandler1->check->message(), $message);
+    }
+
+    /*
+    * Test if the log() function can use its own filters independently of the mockhandlers filters
+    * - Uses two new loggers: $logger and $newlogger
+    *
+    */
+    public function testFiltersOfLoggerWorkWithoutMockHandlerFilter()
+    {
+        // define logger that loads mockhandlers
+        $mockhandlerWihtoutFilter = new MockHandler();
+        $messagefilter[] = new MessageFilter('/filter this out/');
+        $handlers[] = $mockhandlerWihtoutFilter;
+        $logger = new Logger($handlers, null, $messagefilter);
+
+        // random messages
+        $message1 = "this will not be added by Logger, but will be added by Mockhandler without logger";
+        $logMessage1 = new LogMessage($this->level1, $message1);
+
+        // $message1 is filtered out by the logger
+        $logger->log($this->level1, $message1);
+        $data = $logger->serialize();
+        $this->assertStringNotContainsString($data, $message1);
```

Comment:
If you want to test this usecase its more important to test that the message did not reach the mockhandler here right?
Maybe I am missunderstanding.

Reply:
I am slightly confused as well :)